pkg/operations/kubernetesupgrade/upgrader.go

```diff
@@ -19,6 +19,12 @@ type Upgrader struct {
 	kubeConfig string
 }
 
+// UpgradeVM holds VM name and upgrade status
+type UpgradeVM struct {
```

Comment:
I am not clear about the utility of this extra struct. Name can already be derived from: `upgradedAgentsIndex`. If you want change it I would recommend replacing `upgradedAgentsIndex` with this struct.

Reply:
Or to put it another way what is the utility of using a combination of vm name and index to derive information?

Reply:
I need an array of all VMs, both upgraded or to be upgraded. UpgradeVM serves an element in that array. Index is used to create/delete the VMs. Let's discuss offline if the purpose in not clear.

Reply:
Side note the change to naming vms makes it slightly harder to generate vm name based on index, but we must have the data since we are making templates to deploy new vms.